src/System.Drawing.Common/tests/mono/System.Imaging/MetafileTest.cs

```diff
@@ -0,0 +1,705 @@
+        {
+            using (FileStream fs = File.OpenRead(Helpers.GetTestBitmapPath(Emf)))
+            {
+                using (Metafile mf = new Metafile(fs))
+                {
+                    MetafileHeader header1 = mf.GetMetafileHeader();
+                    Check_MetafileHeader_Emf(header1);
+                }
+            }
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void GetMetafileHeader_FromMemoryStream_Emf()
+        {
+            MemoryStream ms;
+            string filename = Helpers.GetTestBitmapPath(Emf);
+            using (FileStream fs = File.OpenRead(filename))
+            {
+                byte[] data = new byte[fs.Length];
+                fs.Read(data, 0, data.Length);
+                ms = new MemoryStream(data);
+            }
+            using (Metafile mf = new Metafile(ms))
+            {
+                MetafileHeader header1 = mf.GetMetafileHeader();
+                Check_MetafileHeader_Emf(header1);
+            }
+            ms.Close();
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Static_GetMetafileHeader_Stream_Null()
+        {
+            Assert.Throws<NullReferenceException>(() => Metafile.GetMetafileHeader((Stream)null));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Static_GetMetafileHeader_Stream()
+        {
+            string filename = Helpers.GetTestBitmapPath(WmfPlaceable);
+            using (FileStream fs = File.OpenRead(filename))
+            {
+                MetafileHeader header = Metafile.GetMetafileHeader(fs);
+                Check_MetafileHeader_WmfPlaceable(header);
+            }
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Static_GetMetafileHeader_Filename_Null()
+        {
+            Assert.Throws<ArgumentNullException>(() => Metafile.GetMetafileHeader((string)null));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Static_GetMetafileHeader_Filename()
+        {
+            string filename = Helpers.GetTestBitmapPath(WmfPlaceable);
+            MetafileHeader header = Metafile.GetMetafileHeader(filename);
+            Check_MetafileHeader_WmfPlaceable(header);
+        }
+    }
+
+    public class MetafileFulltrustTest
+    {
+
+        private Font test_font;
+
+        public MetafileFulltrustTest()
+        {
+            try
+            {
+                test_font = new Font(FontFamily.GenericMonospace, 12);
+            }
+            catch (ArgumentException)
+            {
+            }
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Static_GetMetafileHeader_IntPtr_Zero()
+        {
+            Assert.Throws<ArgumentException>(() => Metafile.GetMetafileHeader(IntPtr.Zero));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Static_GetMetafileHeader_IntPtr()
```

Comment:
I think this is covered by  Static_GetMetafileHeader_IntPtr_ThrowsArgumentException and GetHenhmetafile_ReturnsExpected